### PR TITLE
Fix issues with spec in wafv2 RegularExpressionList

### LINKED ISSUE #1286: WAFv2 error E3002
*cfn-lint version: (`cfn-lint --version`)* 0.26.2

*Description of issue.*

I get an error when running `cfn-lint` against a template with WAFv2 resources. The template provisions the resources in AWS as expected.

```
$ cfn-lint templates/template.yml
E3002 Expecting an object at Resources/AlbWafRegexPatternSet/Properties/RegularExpressionList
templates/template.yml:256:7
```

template snippet:
```
  AlbWafRegexPatternSet:
    Type: 'AWS::WAFv2::RegexPatternSet'
    Properties:
      Description: !Sub "${FullEnv}-${AppName} WAF Regex Pattern Set"
      Name: !Sub "${FullEnv}-${AppName}-regex-pattern-set"
      RegularExpressionList:
        - RegexString: '^.*$'
      Scope: 'REGIONAL'
```

WAFv2 Cloudformation documentation: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-wafv2-regexpatternset.html

### PR DESCRIPTION
**Pull request by kddejong**
*Issue #, if available:*
Fix #1286 
*Description of changes:*
- Patch spec to fix an issue with wafv2 RegularExpressionList

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by kddejong** [comment]
Confirmed that the spec is wrong and that the new changes will both pass cfn-lint and result in a successful deployment.

File: src/cfnlint/data/ExtendedSpecs/all/01_spec_patch.json [hunk 1]
```diff
@@ -108,5 +109,20 @@
     "op": "replace",
     "path": "/PropertyTypes/AWS::Backup::BackupPlan.LifecycleResourceType/Properties/MoveToColdStorageAfterDays/PrimitiveType",
     "value": "Long"
+  },
+  {
+    "op": "replace",
+    "path": "/ResourceTypes/AWS::WAFv2::RegexPatternSet/Properties/RegularExpressionList",
+    "value": {
+      "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-wafv2-regexpatternset-regularexpressionlist.html#cfn-wafv2-regexpatternset-regularexpressionlist-regularexpressionlist",
+      "ItemType": "Regex",
```
Comment (PatMyron) [inline src/cfnlint/data/ExtendedSpecs/all/01_spec_patch.json]:
```suggestion
      "PrimitiveItemType": "String",
```

still getting: `E3002: Expecting an object at Resources/RegexPatternSet/Properties/RegularExpressionList/0`

with this deployable template:

https://github.com/PatMyron/cloud/blob/3a08e527688da6a5b06ff9c2eaa873db31481783/cloudformation/minimal.yaml#L658-L664

```yaml
Resources:
  RegexPatternSet:
    Type: AWS::WAFv2::RegexPatternSet
    Properties:
      Name: Name
      Scope: REGIONAL
      RegularExpressionList:
        - RegularExpression
```

---

Although I'm not sure if the `RegexString` property is also valid or not

**Comment by kddejong** [comment]
@PatMyron are you saying this is wrong too? https://docs.aws.amazon.com/de_de/AWSCloudFormation/latest/UserGuide/aws-properties-wafv2-regexpatternset-regex.html

**Comment by PatMyron** [comment]
@kddejong 

```yaml
Resources:
  RegexPatternSet:
    Type: AWS::WAFv2::RegexPatternSet
    Properties:
      Name: Name
      Scope: REGIONAL
      RegularExpressionList:
        - RegexString: RegularExpression
```

also successfully deploys and now successfully lints, so this is probably fine

**Pull request merged**